assets/js/googlesitekit/widgets/datastore/areas.test.js

```diff
@@ -635,6 +668,106 @@ describe( 'core/widgets Widget areas', () => {
+				registry
+					.dispatch( CORE_WIDGETS )
+					.registerWidgetArea( 'FilteredTestArea', {
+						title: 'Filtered Test Area',
+						subtitle: 'Cool stuff for yoursite.com',
+						style: 'composite',
+						// eslint-disable-next-line no-unused-vars
+						filterActiveWidgets( select, areaWidgets ) {
+							return [];
+						},
+					} );
+				registry
+					.dispatch( CORE_WIDGETS )
+					.assignWidget( 'TestWidget1', 'FilteredTestArea' );
+				registry
+					.dispatch( CORE_WIDGETS )
+					.assignWidget( 'TestWidget2', 'FilteredTestArea' );
+
+				expect(
+					registry
+						.select( CORE_WIDGETS )
+						.isWidgetAreaActive( 'FilteredTestArea' )
+				).toBe( false );
+			} );
+
+			it( 'returns false when only one widget remains after `filterActiveWidgets`', () => {
+				registry
+					.dispatch( CORE_WIDGETS )
+					.registerWidgetArea( 'SingleWidgetTestArea', {
+						title: 'Single Widget Test Area',
+						subtitle: 'Cool stuff for yoursite.com',
+						style: 'composite',
+						filterActiveWidgets( select, areaWidgets ) {
+							return areaWidgets.length === 1 ? [] : areaWidgets;
+						},
+					} );
+				registry
+					.dispatch( CORE_WIDGETS )
+					.assignWidget( 'TestWidget1', 'SingleWidgetTestArea' );
+
+				expect(
+					registry
+						.select( CORE_WIDGETS )
+						.isWidgetAreaActive( 'SingleWidgetTestArea', {
+							modules: [ 'test-module-1' ],
+						} )
+				).toBe( false );
+			} );
+
+			it( 'returns true when `filterActiveWidgets` keeps widgets', () => {
+				registry
+					.dispatch( CORE_WIDGETS )
+					.registerWidgetArea( 'FilteredTestArea', {
+						title: 'Filtered Test Area',
+						subtitle: 'Cool stuff for yoursite.com',
+						style: 'composite',
+						filterActiveWidgets( select, areaWidgets ) {
+							return areaWidgets.length === 1 ? [] : areaWidgets;
+						},
+					} );
+				registry
+					.dispatch( CORE_WIDGETS )
+					.assignWidget( 'TestWidget1', 'FilteredTestArea' );
+				registry
+					.dispatch( CORE_WIDGETS )
+					.assignWidget( 'TestWidget2', 'FilteredTestArea' );
+
+				expect(
+					registry
+						.select( CORE_WIDGETS )
+						.isWidgetAreaActive( 'FilteredTestArea' )
+				).toBe( true );
+			} );
+
+			it( 'returns true when `filterActiveWidgets` does not alter widgets', () => {
+				registry
+					.dispatch( CORE_WIDGETS )
+					.registerWidgetArea( 'UnfilteredTestArea', {
+						title: 'Unfiltered Test Area',
+						subtitle: 'More cool stuff for yoursite.com',
+						style: 'composite',
+						filterActiveWidgets( select, areaWidgets ) {
+							return areaWidgets;
+						},
+					} );
+				registry
+					.dispatch( CORE_WIDGETS )
+					.assignWidget( 'TestWidget1', 'UnfilteredTestArea' );
+				registry
+					.dispatch( CORE_WIDGETS )
+					.assignWidget( 'TestWidget2', 'UnfilteredTestArea' );
+
+				expect(
+					registry
+						.select( CORE_WIDGETS )
+						.isWidgetAreaActive( 'UnfilteredTestArea' )
+				).toBe( true );
+			} );
```

Comment:
This is sort of an oddly-named test, because really we should be comparing the input widgets and output. If we supplied no widgets and didn't filter them, we would actually return `false`, so the test spec is misleading.

I'm not sure this test is really needed, given _omitting_ the filter already does this 🙂

Reply:
Good shout. The reason to keep this test will be if we want to explicitly ensure that the behavior of the `filterActiveWidgets` function is correct, even when it's implemented to return the widgets unchanged. Essentially, we are verifying that introducing `filterActiveWidgets` does not inadvertently break existing functionality when it does not alter widgets. However, the name of the test indeed is misleading. I will change it to `returns true when `filterActiveWidgets` is implemented but returns the widgets unchanged and widgets are active`. Does that sound good?

Reply:
Sounds good 😄